scaleway/data_source_ipam_ip.go

```diff
@@ -0,0 +1,97 @@
+package scaleway
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/hashicorp/go-cty/cty"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	ipam "github.com/scaleway/scaleway-sdk-go/api/ipam/v1alpha1"
+	"github.com/scaleway/scaleway-sdk-go/scw"
+)
+
+func dataSourceScalewayIPAMIP() *schema.Resource {
+	return &schema.Resource{
+		ReadContext: dataSourceScalewayIPAMIPRead,
+		Schema: map[string]*schema.Schema{
+			// Input
+			"private_network_id": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"resource_id": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"mac_address": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"type": {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "IP Type (ipv4, ipv6)",
+			},
+			"region": regionSchema(),
+
+			// Computed
+			"address": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+		},
+	}
+}
+
+func dataSourceScalewayIPAMIPRead(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	api, region, err := ipamAPIWithRegion(d, meta)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	req := &ipam.ListIPsRequest{
+		Region:           region,
+		OrganizationID:   nil,
+		PrivateNetworkID: expandStringPtr(d.Get("private_network_id")),
+		SubnetID:         nil,
+		Attached:         nil,
+		ResourceID:       expandStringPtr(d.Get("resource_id")),
+		ResourceType:     ipam.ResourceTypeUnknownType,
+		MacAddress:       expandStringPtr(d.Get("mac_address")),
+		ResourceName:     nil,
+		ResourceIDs:      nil,
```

Comment:
```suggestion
		PrivateNetworkID: expandStringPtr(d.Get("private_network_id")),
		ResourceID:       expandStringPtr(d.Get("resource_id")),
		ResourceType:     ipam.ResourceTypeUnknownType,
		MacAddress:       expandStringPtr(d.Get("mac_address")),
```

Simplify the code

Reply:
This show clearly the missing filters, this is as efficient as a TODO that I will add to it